### DISCUSSION TITLE
Pagination and calling REST APIs

### DISCUSSION THREAD

**Discussion by alvinashcraft** [ChilliCream/hotchocolate#4161] [Q&A] [answered]
We have a GraphQL server that is calling several REST endpoints to aggregate some data for the client.

The server calls the first REST API to get an initial list of items and then makes a couple of other REST calls in parallel to get more details. We would like to use pagination with GraphQL and to be able to pass the paging request through to that first REST endpoint.

Is there anything in GraphQL or Hot Chocolate to facilitate this?

Adding as Q&A, but maybe this could be a feature idea if it's not available yet.

**Comment by tobias-tengler** [comment, marked as answer]
I'm not sure if I understand your use-case correctly, but you can customize the pagination resolver, i.e. access the paging arguments and forward them to your REST service. This is covered by the documentation: https://chillicream.com/docs/hotchocolate/fetching-data/pagination/#customization

**Reply by alvinashcraft** [reply]
Thank you. I think this method would cover our use case. We'll give it a try. Here's an idea of what we're trying to do:

// call GraphQL (pass paging info)
GetWidgetInfo(int customerId)

// GraphQL server calls REST endpoint 1 (pass thru paging info from client)
var orders = GetWidgetOrders(int customerId)

// GraphQL gets more details (two calls in parallel for all orders in current page)
var widgetDetails = GetWidgetInfo(orders)
var orderDetails = GetOrderInfo(orders)

// code to stitch details to orders and return everything to caller

**Reply by michaelstaib** [reply]
We did a lot of work on the cursor pagination in V12 to make it much easier to plugin a custom pagination provider. RC.1 will be on NuGet tonight.
